File: CBLClient/Apps/JavaRequestHandlers/src/main/java/com/couchbase/lite/mobiletestkit/legacy/SessionAuthenticatorRequestHandler.java
package com.couchbase.lite.mobiletestkit.legacy;


import com.couchbase.lite.mobiletestkit.Args;
import com.couchbase.lite.SessionAuthenticator;


public class SessionAuthenticatorRequestHandler {
    /* ------------------------ */
    /* - SessionAuthenticator - */
    /* ------------------------ */

    public SessionAuthenticator create(Args args) {
        String sessionId = args.getString("sessionId");
        String cookieName = args.getString("cookieName");
        if (cookieName == null) {
            return new SessionAuthenticator(sessionId);
        }
        return new SessionAuthenticator(sessionId, cookieName);
    }

    public String getSessionId(Args args) {
        SessionAuthenticator session = args.get("session", SessionAuthenticator.class);
        return session.getSessionID();
    }

    public String getCookieName(Args args) {
        SessionAuthenticator session = args.get("session", SessionAuthenticator.class);
        return session.getCookieName();
    }
}

File: CBLClient/Apps/JavaRequestHandlersKotlin/src/main/java/com/couchbase/mobiletestkit/javacommon/RequestHandler/SessionAuthenticatorRequestHandler.kt
package com.couchbase.mobiletestkit.javacommon.RequestHandler

import com.couchbase.lite.SessionAuthenticator
import com.couchbase.mobiletestkit.javacommon.Args

class SessionAuthenticatorRequestHandler {
    /* ------------------------ */ /* - SessionAuthenticator - */ /* ------------------------ */
    fun create(args: Args): SessionAuthenticator {
        val sessionId = args.get<String>("sessionId")
        val cookieName = args.get<String>("cookieName") ?: return SessionAuthenticator(sessionId)
        return SessionAuthenticator(sessionId, cookieName)
    }

    fun getSessionId(args: Args): String {
        val session = args.get<SessionAuthenticator>("session")
        return session.sessionID
    }

    fun getCookieName(args: Args): String? {
        val session = args.get<SessionAuthenticator>("session")
        return session.cookieName
    }
}